src/asl.ts

```diff
@@ -4871,13 +5016,331 @@ export namespace ASLGraph {
           }),
     };
   };
+
+  /**
+   * Compare any two {@link ASLGraph.Output} values.
+   */
+  export const compareOutputs = (
+    leftOutput: ASLGraph.Output,
+    rightOutput: ASLGraph.Output,
+    operator: ASL.ValueComparisonOperators | "!=" | "!=="
```

Comment:
I don't like these adhoc types. Why is `!=` and `!==` not a value comparison operator?

Reply:
OK, I just added them to the table and set all of the values to undnefined.

Reply:
Why not implement them?

Reply:
ASL doesn't have a StringNotEquals (the table is a map between operator and the ASL name (StringEquals, NumberLessThan, etc).

`!==` is `Not(*Equals)`.

Reply:
Why not implement them as functions? There are a bunch of functions referencing this map as the authoritative source of supported operators. It's difficult to follow.

Reply:
Was trying to generalize the handling of the matrix of operator, type, and path or constant based. I'll take a pass at this in another PR.